### PR TITLE
v1 transport pipeline and AMQP spec.

### PR DESCRIPTION
**Pull request by CIPop**
Specifications for Transport Pipeline and AMQP.

### CODE REVIEW

File: iothub/device/devdoc/architecture.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+The `DefaultDelegatingHandler` class implements the delegating portion from the "chain of responsibility" pattern by routing messages to the next handler in the chain (`InnerHandler`).
+
+Functional requirements:
+1. Implements `IDisposable`: throws when pipeline use-after-dispose is detected.
+1. Implements the `IDelegatingHandler` internal IoT Hub protocol methods (see the Architecture diagram).
+
+### RetryDelegatingHandler
+
+`RetryDelegatingHandler` is responsible with keeping the client in sync with the state requested by the application. This class is also responsible with applying the operation retry policy.<br>
```
Comment (CIPop) [inline iothub/device/devdoc/architecture.md]: Should split into State and Retry handlers.

File: doc/devguide.md [hunk 2]
```diff
@@ -119,7 +122,7 @@ Contains common code shared between multiple components. This doesn't generate a
 `/test` contains end-to-end tests run before each PR integration by our internal CI system.
 `/stress` contains SDK stress-test applications.
 
-### /jenkins
+### /jenkins and /vsts
 
 Contains scripts used by our internal Continuous Integration system (Jenkins).
```
Comment (abhipsaMisra) [inline doc/devguide.md]: Do we still use Jenkins for CI?

File: iothub/device/devdoc/amqpstack.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+1. If the graceful close causes any exceptions, the AMQP connection will be `Abort`ed.
+1. Changing the active state of the connection needs to be performed under a lock so that the `AmqpIoTConnectionPool` can be sure of the reported state.
+
+_Note:_ The `AmqpIoTConnection` will not manage the session lifetime although it will register for `AmqpSession.Closed` events.
+
+### X509 Authentication
+Clients using X509 authentication cannot use multiplexing. Internally, a default X509 connection pool will generate these connections.
+
+### CBS Authentication
+Clients using SAS token will need to create and maintain a `TokenRefresher` object. `InternalClient` will maintain ownership of this object. A disposed `TokenRefresher` will indicate that the client no longer needs to be authenticated.
```
Comment (CIPop) [inline iothub/device/devdoc/amqpstack.md]: Batch `TokenRefresh`s:
- refresh tokens within 1-5seconds
- refresh maximum 1k tokens / second

File: doc/devguide.md (hunk 2 above)
Comment (CIPop) [reply doc/devguide.md]: No: good point - will remove the Jenkins ref.

File: iothub/device/devdoc/architecture.md (hunk 1 above)
Comment (CIPop) [reply iothub/device/devdoc/architecture.md]: We've decided to keep things as they are for now.

File: iothub/device/devdoc/amqpstack.md (hunk 3 above)
Comment (CIPop) [reply iothub/device/devdoc/amqpstack.md]: Modified spec.

**Pull request merged**